### PR TITLE
Fix up Dir.glob for windows

### PR DESCRIPTION
**Pull request by jaym**
Dir.glob does not work on windows if a symlinked directory is in the pattern. This is most likely the issue being seen in https://github.com/test-kitchen/test-kitchen/issues/1238

This patch takes the approach of changing to the directory before doing any globbing.

### CODE REVIEW

File: lib/berkshelf/file_syncer.rb [hunk 1]
```diff
@@ -102,42 +108,19 @@ def sync(source, destination, options = {})
...
-      # @param [String] path
-      #   the path to get relative with
-      # @param [String] parent
-      #   the parent where the path is contained (hopefully)
-      #
-      # @return [String]
-      #
-    def relative_path_for(path, parent)
-      Pathname.new(path).relative_path_from(Pathname.new(parent)).to_s
-    end
```
Comment (lamont-granquist) [inline lib/berkshelf/file_syncer.rb]: 👍

**Comment by lamont-granquist** [comment]
i knew there was a reason i tended to do this, i had no idea it was because $windows..

**Comment by lamont-granquist** [comment]
i knew there was a reason i tended to do this, i had no idea it was because $windows..

File: lib/berkshelf/file_syncer.rb [hunk 2]
```diff
@@ -61,21 +68,20 @@ def sync(source, destination, options = {})
       # let glob expand the source directory in case it is an abbreviated windows
       # user directory: C:/Users/MATTWR~1/AppData/Local/Temp
       # so that it matches the parent of source_files
-      source = glob(source).first
+      # source = glob(source).first
```
Comment (thommay) [inline lib/berkshelf/file_syncer.rb]: please to remove the commented out line

**Pull request merged**